app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -292,8 +307,13 @@ function FormSteps({
     }, [] as FormStepError<Record<string, Error>>[]);
   }
 
-  // An empty steps array is allowed, in which case there is nothing to render.
-  if (!step) {
+  if (step) {
+    useStepTitle(step, titleFormat);
+    useTrackStepEvents(step);
+    useDidUpdateEffect(() => onStepChange(step), [step]);
+    useDidUpdateEffect(onPageTransition, [step]);
```

Comment:
Hooks can't be conditional.

>Don’t call Hooks inside loops, conditions, or nested functions.

https://reactjs.org/docs/hooks-rules.html

We should consider updating the hooks to allow for undefined, and update the logic to deal with it accordingly.